src/ucm/cuda/cudamem.c

```diff
@@ -286,22 +293,32 @@ static ucs_status_t ucm_cudamem_install(int events)
         goto out_unlock;
     }
 
-    status = ucm_cuda_install_hooks(ucm_cuda_driver_funcs, &used_reloc,
-                                    "driver");
+    /* With bistro, hooking the driver API will cover the runtime API too.
+     * With reloc, we need to hook both the driver and runtime API.
+     * In both cases, we return success as long as the runtime API is hooked
+     * successfully, because most applications use the runtime API only.
+     */
+    status = ucm_cuda_install_hooks(ucm_cuda_driver_funcs, "driver");
+    if (status == UCS_OK && ucm_cuda_hook_mode() == UCM_MMAP_HOOK_BISTRO) {
+        ucm_cudamem_installed = 1;
+        goto out_unlock;
+    }
+
     if (status != UCS_OK) {
         ucm_warn("failed to install cuda memory hooks on driver API");
-    } else if (!used_reloc) {
+    }
+
+    status = ucm_cuda_install_hooks(ucm_cuda_runtime_funcs, "runtime");
+    if (status == UCS_OK) {
         ucm_cudamem_installed = 1;
-    } else if (status == UCS_OK) {
-        /* Failed to install bistro hooks on all driver APIs, so need to install
-           hooks on runtime APIs. */
-        status = ucm_cuda_install_hooks(ucm_cuda_runtime_funcs, &used_reloc,
-                                        "runtime");
-        if (status == UCS_OK) {
-            ucm_cudamem_installed = 1;
-        } else {
-            ucm_warn("failed to install cuda memory hooks on runtime API")
-        }
+        goto out_unlock;
+    }
```

Comment:
seems this logic is incorrect

(1) is bistro fails with driver funcs,  it is not sufficient with just runtime funcs bistro because the application might be using driver app
(2) with "reloc",  both driver and runtime func needs to be successfully installed to consider ucm_cudamem_installed = 1